Java/src/main/java/com/gildedrose/StoredItem.java

```diff
@@ -0,0 +1,63 @@
+package com.gildedrose;
+
+public class StoredItem extends Item implements NamedItem, ItemWithQuality, LegacyItem {
+
+    public StoredItem(String name, int sellIn, int quality) {
+        super(name, sellIn, quality);
+    }
+
+    public StoredItem(Item item) {
+        super(item.name, item.sellIn, item.quality);
+	}
+
+    @Override
+    public void decreaseQuality() {
+        this.quality = this.quality - 1;
+    }
+
+    @Override
+    public void increaseQuality() {
+        this.quality = this.quality + 1;
+    }
+
+    @Override
+    public void loseAllQuality() {
+        this.quality = 0;
+    }
+
+    @Override
+    public boolean isNamed(String name) {
+        return this.name.equals(name);
+    }
+
+    @Override
+    public boolean isNotNamed(String name) {
+        return !this.name.equals(name);
+    }
+
+    @Override
+    public Item convertBack() {
```

Comment:
StoredItem should be compatible with Item and needs no conversion.

Reply:
yep, brainfart